tests/components/mysensors/test_device_tracker.py

```diff
@@ -0,0 +1,55 @@
+"""Provide tests for mysensors device tracker platform."""
+from __future__ import annotations
+
+from collections.abc import Callable
+
+from mysensors.sensor import Sensor
+
+from homeassistant.components.device_tracker import ATTR_SOURCE_TYPE, SourceType
+from homeassistant.const import ATTR_LATITUDE, ATTR_LONGITUDE, STATE_NOT_HOME
+from homeassistant.core import HomeAssistant
+
+
+async def test_gps_sensor(
+    hass: HomeAssistant,
+    gps_sensor: Sensor,
+    receive_message: Callable[[str], None],
+) -> None:
+    """Test a gps sensor."""
+    entity_id = "device_tracker.gps_sensor_1_1"
+    altitude = 0
+    latitude = "40.742"
+    longitude = "-73.989"
+    message_string = f"1;1;1;0;49;{latitude},{longitude},{altitude}\n"
+
+    receive_message(message_string)
+    # the integration adds multiple jobs to do the update currently
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
```

Comment:
Should this be factored out into something like an `await_all_jobs` function, which can be reused? Three consecutively repeated side-effects here seems like it should be abstracted away.

Reply:
https://github.com/home-assistant/core/pull/84580#discussion_r1057325787